File: Skills/goal-decomposition/scripts/build_confirmation_prompt.py
#!/usr/bin/env python3
import argparse
from pathlib import Path


def main():
    parser = argparse.ArgumentParser()
    parser.add_argument("--plan-file", required=True)
    parser.add_argument("--max-chars", type=int, default=12000)
    args = parser.parse_args()

    plan_file = Path(args.plan_file).resolve()
    if not plan_file.exists():
        raise SystemExit(f"Plan file not found: {plan_file}")

    content = plan_file.read_text(encoding="utf-8")
    content = content.replace("\r\n", "\n").strip()

    suffix = "\n\n[文档内容过长，已截断显示。完整文件见上方路径。]"
    if len(content) > args.max_chars:
        keep = max(0, args.max_chars - len(suffix))
        content = content[:keep] + suffix

    prompt = (
        f"请先审查以下拆解文档，再选择操作。\n"
        f"文件路径：{plan_file}\n\n"
        f"--- 文档内容开始 ---\n"
        f"{content}\n"
        f"--- 文档内容结束 ---\n\n"
        f"请选择：确认 或 修改"
    )
    print(prompt)


if __name__ == "__main__":
    main()

File: Skills/goal-decomposition/scripts/build_confirmation_prompt.js
#!/usr/bin/env bun
import path from "node:path"
import { existsSync, readFileSync } from "node:fs"

function arg(name, fallback = "") {
  const idx = process.argv.indexOf(name)
  return idx >= 0 ? process.argv[idx + 1] || "" : fallback
}

const planFileRaw = arg("--plan-file")
const maxChars = Number(arg("--max-chars", "12000"))
if (!planFileRaw) throw new Error("--plan-file is required")
const planFile = path.resolve(planFileRaw)
if (!existsSync(planFile)) throw new Error(`Plan file not found: ${planFile}`)
let content = readFileSync(planFile, "utf8").replace(/\r\n/g, "\n").trim()
const suffix = "\n\n[文档内容过长，已截断显示。完整文件见上方路径。]"
if (content.length > maxChars) {
  const keep = Math.max(0, maxChars - suffix.length)
  content = `${content.slice(0, keep)}${suffix}`
}
const prompt = `请先审查以下拆解文档，再选择操作。\n文件路径：${planFile}\n\n--- 文档内容开始 ---\n${content}\n--- 文档内容结束 ---\n\n请选择：确认 或 修改`
console.log(prompt)
